activerecord/test/cases/persistence_test.rb

```diff
@@ -873,6 +874,22 @@ def test_destroy_for_a_failed_to_destroy_cpk_record
     end
   end
 
+  def test_destroy_after_restart_parent_transaction_rollback
+    MyPost.create!
+    assert_equal 1, MyPost.count
+    MyPost.test_restart_parent_transaction
+    assert_equal 0, MyPost.count
+  end
```

Comment:
I noticed some issues with the tests:
- It is preferred to reuse existing models. 
- The initial quantity of records cannot be guaranteed because of fixtures. For example `Topic.count` currently returns 5 in your test.
- Having to jump back and forth between the model and the test muddies what is being test here. It took me a minute to understand the issue. 
- I don't understand what is being tested from the test name.  Adding a reference to the issue in comment could help future maintainer understand the context of this test.

Here's how I would write the test to account for all 4:

```suggestion
  def test_destroy_after_transaction_rollback
    # Regression test for #50338
    topic = Topic.create!
    assert_difference("Topic.count", -1) do
      begin
        ActiveRecord::Base.transaction do
          topic.destroy
          raise StandardError.new
        end
      rescue
        topic.destroy
      end
    end
  end
```